base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1259 @@
+   use MAPL_ShmemMod
+   use mapl_ErrorHandlingMod
+   use MAPL_Constants
+   use ESMF
+   use pFIO
+   use MAPL_CommsMod
+   use, intrinsic :: iso_fortran_env, only: REAL32
+   use, intrinsic :: iso_fortran_env, only: REAL64
+   implicit none
+   private
+
+   public :: SwathGridFactory
+
+   integer, parameter :: NUM_DIM = 2
+   
+   type, extends(AbstractGridFactory) :: SwathGridFactory
+      private
+      character(len=:), allocatable :: grid_name
+      character(len=:), allocatable :: grid_file_name      
+      ! Grid dimensions
+      integer :: im_world = MAPL_UNDEFINED_INTEGER
+      integer :: jm_world = MAPL_UNDEFINED_INTEGER
+      integer :: lm = MAPL_UNDEFINED_INTEGER
+      ! prepare to delete
+      !real(kind=REAL64), allocatable :: lon_centers(:)
+      !real(kind=REAL64), allocatable :: lat_centers(:)
+      logical :: force_decomposition = .false.
+
+      ! Domain decomposition:
+      integer :: nx = MAPL_UNDEFINED_INTEGER
+      integer :: ny = MAPL_UNDEFINED_INTEGER
+      integer, allocatable :: ims(:)
+      integer, allocatable :: jms(:)
+      ! Grid conventions:
+      ! Regional vs global: ?? not needed
+      type (RealMinMax) :: lon_range = RealMinMax(MAPL_UNDEFINED_REAL,MAPL_UNDEFINED_REAL)
+      type (RealMinMax) :: lat_range = RealMinMax(MAPL_UNDEFINED_REAL,MAPL_UNDEFINED_REAL)
+      ! Used for halo
+      type (ESMF_DELayout) :: layout
+      logical :: initialized_from_metadata = .false.
+   contains
+      procedure :: make_new_grid
+      procedure :: create_basic_grid
+      procedure :: add_horz_coordinates_from_file
+      procedure :: init_halo
+      procedure :: halo
+
+      procedure :: initialize_from_file_metadata
+      procedure :: initialize_from_config_with_prefix
+      procedure :: initialize_from_esmf_distGrid
+
+      procedure :: equals
+      procedure :: check_and_fill_consistency
+      procedure :: generate_grid_name
+      procedure :: to_string
+
+      procedure :: append_metadata
+      procedure :: get_grid_vars
+      procedure :: get_file_format_vars
+      procedure :: append_variable_metadata
+      procedure :: check_decomposition
+      procedure :: generate_newnxy
+      procedure :: generate_file_bounds
+      procedure :: generate_file_corner_bounds
+      procedure :: generate_file_reference2D
+      procedure :: generate_file_reference3D
+      procedure :: decomps_are_equal
+      procedure :: physical_params_are_equal
+   end type SwathGridFactory
+
+   character(len=*), parameter :: MOD_NAME = 'MAPL_SwathGridFactory::'
+
+   interface SwathGridFactory
+      module procedure SwathGridFactory_from_parameters
+   end interface SwathGridFactory
+
+   interface set_with_default
+      module procedure set_with_default_integer
+      module procedure set_with_default_real
+      module procedure set_with_default_real64
+      module procedure set_with_default_character
+      module procedure set_with_default_bounds
+   end interface set_with_default
```

Comment:
Note to myself.   It might make sense to introduce a MAPL module to support with optional things.  Clearly these are useful here and in the LatLon grid factory.  My bet is that the integer cases are useful in many locations.